keras_nlp/models/task.py

```diff
@@ -217,60 +223,111 @@ def from_preset(
                 "constructor with a `backbone` argument. "
                 f"Received: backbone={kwargs['backbone']}."
             )
-        preset_cls = check_config_class(preset)
+
+        # Check if we should load a `task.json` directly.
+        load_task_config = False
+        if check_file_exists(preset, TASK_CONFIG_FILE):
+            task_preset_cls = check_config_class(preset, TASK_CONFIG_FILE)
+            if issubclass(task_preset_cls, cls):
+                load_task_config = True
+        if load_task_config:
+            # Task case.
+            task_preset_cls = check_config_class(preset, TASK_CONFIG_FILE)
+            task = load_serialized_object(preset, TASK_CONFIG_FILE)
+            if load_weights:
+                task.load_task_weights(get_file(preset, TASK_WEIGHTS_FILE))
+                task.backbone.load_weights(get_file(preset, MODEL_WEIGHTS_FILE))
```

Comment:
I think we might need that `jax_memory_cleanup` call before the load weights lines? (also it's a good idea to keep that split out, because ideally we work that into core Keras someday)

Reply:
Right! Added!